internal/config/config.go

```diff
@@ -58,14 +59,17 @@ type Config struct {
 	Version string `yaml:"version" validate:"eq=v2"`
 	// description: |
 	//   Machine image used to create Constellation nodes.
-	Image string `yaml:"image" validate:"required"`
+	Image string `yaml:"image" validate:"required,version_compatibility"`
 	// description: |
 	//   Size (in GB) of a node's disk to store the non-volatile state.
 	StateDiskSizeGB int `yaml:"stateDiskSizeGB" validate:"min=0"`
 	// description: |
-	//   Kubernetes version to be installed in the cluster.
+	//   Kubernetes version to be installed into the cluster.
 	KubernetesVersion string `yaml:"kubernetesVersion" validate:"supported_k8s_version"`
 	// description: |
+	//	 Microservice version to be installed into the cluster. Optional until v2.7. Defaults to the version of the CLI.
```

Comment:
Should we use this string to inform about allowed version ranges? Or should `upgrade check` only ever modify this value?

Reply:
I think I don't understand what you mean. This comment ends up in the default config as a comment. And the version number should hint at the fact that setting the `microserviceVersion` value in the config is optional for v2.6. For compatibility reasons. Starting with 2.7 the values _has_ to be set. I extended the comment a bit.

The value is a bit redundant as a given CLI in version X is only able to upgrade the service versions to the one version included in CLI X. So its more informational to the user than anything else. 

I don't think this comment should mention available versions as we would have to update the comment manually.
`upgrade check` displays the available versions and writes them to the config if `--write-config` is set.

Reply:
I was thinking along the lines of: `This version should not surpess the current cluster version by more than one. See <link to our upgrade docs>` 

In case the user wants to manually set a version, instead of using `upgrade check` they get a hint about accepted values and further documentation.

Reply:
In case you put values into the field that don't adhere to the compatibility constraints you will get pretty precise error messages during config validation [1].

I will add links to the docs once I wrote them for the respective fields. 

[1] https://github.com/edgelesssys/constellation/pull/1109/files#diff-da15a1a7d9e1c206009e1cec96dc5f52b1ad2fc84ca481a477bc73716866c3c1R315-R319